src/index.js

```diff
@@ -1,18 +1,23 @@
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
 import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
 import { ethers } from 'ethers';
-import React, { useEffect } from 'react';
+import React, { useCallback, useEffect } from 'react';
 import ReactDOM from 'react-dom/client';
 import { App } from './App';
 import { SynthetixProvider, useSynthetix } from './useSynthetix';
 import { restoreToken } from './utils';
 
 import './main.css';
 
+export const OP_SEPOLIA_CHAIN_ID = '0xaa37dc';
+
 const queryClient = new QueryClient();
 
 function WalletWatcher({ children }) {
   const [, updateSynthetix] = useSynthetix();
+  const handleChainChanged = useCallback(() => {
+    window.location.reload();
```

Comment:
there is no need to reload page when chain changes.
Make sure you add chainId to cache string to all react-queries, so after chain id change all queries will be invalidated and re-fetched

Reply:
Remove 'chainChanged' event handler